services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/structure/DigitalCertificatesDirectory.java

```diff
@@ -0,0 +1,66 @@
+package app.coronawarn.server.services.distribution.dgc.structure;
+
+import app.coronawarn.server.common.protocols.internal.dgc.ValueSets;
+import app.coronawarn.server.common.shared.collection.ImmutableStack;
+import app.coronawarn.server.services.distribution.assembly.component.CryptoProvider;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.ArchiveOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.decorator.signing.DistributionArchiveSigningDecorator;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.IndexDirectoryOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.file.FileOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig.DigitalGreenCertificate;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DigitalCertificatesDirectory extends IndexDirectoryOnDisk<String> {
+
+  private static final Logger logger = LoggerFactory.getLogger(DigitalCertificatesDirectory.class);
+
+  public static final String EXPORT_BINARY_FILENAME = "export.bin";
+
+  private final DistributionServiceConfig distributionServiceConfig;
+
+  private final DigitalGreenCertificate dgcConfig;
+
+  private final CryptoProvider cryptoProvider;
+
+  private ValueSets valueSets;
+
+  /**
+   * Constructs a {@link IndexDirectoryOnDisk} instance that represents a directory, containing an index in the form of
```

Comment:
```suggestion
   * Constructs a {@link IndexDirectoryOnDisk} instance that represents a directory for a digital green certificates and writes valuesets to all supported countries by the dgc.
```

Maybe we can have a more meaningful comment. Sry for nitpicking here